### ISSUE TITLE
Retrieving image without file extension in URL fails

### ISSUE THREAD

**Issue by rijkvanzanten** [directus/directus#1852]
## Version Info
  - Directus version and branch: develop latest (https://github.com/directus/directus/commit/a5bcc69c9343413c7f6c0c4cd028c08d8938b1d2)
  - PHP version: 7.1.6

## Expected Behavior
Retrieving an image through an URL gets the image.

## Actual Behavior
Trying to save the image throws an error
```
Server Error
Undefined index: extension
```

## Steps to Reproduce
 1. Upload a new file
 2. Use an image url without an extension (like `https://unsplash.it/500/500`

**Comment by parnas** [comment]
Even worse, we store image uploads with extension of the original file. Which means, that it is possible, for example, to upload php file as image.

And yes, we do have .htaccess in uploads that does
```
<FilesMatch "\.(php|phps|php5|htm|shtml|xhtml|cgi.+)$">
    ForceType text/plain
</FilesMatch>
```
but that is only helping against listed extensions (I've seen php also setup as .phtml and .php4, for example).

And that is just the one type of attack. It's also possible to upload an html file, that would be served from the directus domain and, thus, would have access to directus domain cookies.

Basically, if I have can upload files, I can get session info of another directus user on the same domain, by sending that user a link to the uploaded file.

The files should never be saved with the original extension, instead extensions should be assigned (something like https://github.com/ralouphie/mimey could be used to make defining extensions easier).

Also, we already do getimagesize, but we should also check if width and height is zero, and don't accept a file like that. 

@WellingGuzman @benhaynes

**Comment by WellingGuzman** [comment]
We should discuss this in a separate issue, about how we treat uploading php files and such and a way to reproduce and how avoid each possible conditions.

---

Fixed by https://github.com/directus/directus/commit/4098667afce9571c0d047be02a210b25a3d37e61

**Comment by rijkvanzanten** [comment]
In what issue? Did you open a new one? @wellingguzman

**Comment by WellingGuzman** [comment]
@rijkvanzanten I missed the pointed out we should talk what @parnas, was talking about uploading php files and the other extensions that I can run in another ticket, as it's a separate security issue. which is how we prevent future extensions.

I'm not too much concern as we can add it as they are coming up, and if they can set up weird extensions that out of hands. Again if there's a better way to handle this, I would love to hear it.

The problem is that we wanted to serve the file as the original name + extension, if we can skip that we can generate random extensions or new extensions such as change `.php` to `.php.txt` and similar.
